File: EduQuiz/SharedUserControls/AppBarButtOpenWeb.vb

Imports pkar.UI.Extensions


#If Not NETFX_CORE And Not PK_WPF Then
' assume WinUI3
Imports Microsoft.UI.Xaml
#End If


Public Class AppBarButtOpenWeb
#If PK_WPF Then
    Inherits AppBarButton
#Else
    Inherits Controls.AppBarButton
#End If

    Public Property Uri As Uri

    Public Sub New()
        MyBase.New

#If PK_WPF Then
        Icon = "Controls.Symbol.Globe"
#Else
        Icon = New Controls.SymbolIcon(Controls.Symbol.Globe)
#End If
        Label = "Open Web"

        AddHandler Me.Click, AddressOf PojdzDoWWW

    End Sub

    Private Sub PojdzDoWWW(sender As Object, e As RoutedEventArgs)
        Uri.OpenBrowser
    End Sub
End Class

File: QuizKursUno/QuizKursUno/SharedUserControls/AppBarButtOpenWeb.cs

using pkar.UI.Extensions;

namespace QuizKursUno;
public partial class AppBarButtOpenWeb : AppBarButton
{
    public Uri Uri { get; set; }

    AppBarButtOpenWeb() : base()
    {
        Icon = new SymbolIcon(Symbol.Globe);
        Label = "Open Web";
        this.Uri = new Uri("http://");  // że niby nullable, do ustawienia w ctor, i w ogóle
        this.Click += PojdzDoWWW;
    }

    private void PojdzDoWWW(Object sender, RoutedEventArgs e)
    {
        this.Uri.OpenBrowser();
    }
}
